modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/GnirsEpics.scala

```diff
@@ -174,14 +174,30 @@ class GnirsEpics(epicsService: CaService, tops: Map[String, String]) {
 
   def decker: Option[String] = Option(state.getStringAttribute("decker").value)
 
-  def centralWavelength: Option[Double] = Option(state.getStringAttribute("centralWavelength").value).map(_.toDouble)
+  def centralWavelength: Option[Double] = Option(state.getDoubleAttribute("centralWavelength").value).map(_.toDouble)
 
-  def gratingTilt: Option[Double] = Option(state.getStringAttribute("grattilt").value).map(_.toDouble)
+  def gratingTilt: Option[Double] = Option(state.getDoubleAttribute("grattilt").value).map(_.toDouble)
 
   def nirscc: Option[String] = Option(state.getStringAttribute("nirscc").value)
 
   def gratingOrder: Option[Int] = Option(state.getIntegerAttribute("gratord").value).map(_.toInt)
 
+  def filter1Eng: Option[Int] = Option(state.getIntegerAttribute("fw1_eng").value).map(_.toInt)
```

Comment:
I have the impression that reading the value is a side effect, as we read it from the EPICS channel. Is that correct?

Reply:
The value is cached, so there is no network activity in this call. But, it breaks referential transparency.